library/utils.c

```diff
@@ -141,30 +140,43 @@ void ziti_log_init(uv_loop_t *loop, int level, log_writer log_func) {
     }
 }
 
-void ziti_log_set_level(int level) {
+void ziti_log_set_level(int level, const char *marker) {
     if (level == ZITI_LOG_DEFAULT_LEVEL) {
-        char *lvl = getenv("ZITI_LOG");
-        if (lvl != NULL) {
-            ziti_log_lvl = (int) strtol(lvl, NULL, 10);
-            int num_levels = sizeof(level_labels) / sizeof(const char *);
-            if (ziti_log_lvl < 0 || ziti_log_lvl >= num_levels) {
-                ziti_log_lvl = INFO;
+        if (marker) {
+            model_map_remove(&log_levels, marker);
+        }
+    } else {
+        if (marker) {
+            model_map_set(&log_levels, marker, (void *) (uintptr_t) level);
+            if (strcmp(marker, UV_MBED_MODULE) == 0) {
+                uv_mbed_set_debug(level, uv_mbed_logger);
             }
         } else {
-            ziti_log_lvl = INFO;
+            ziti_log_lvl = level;
         }
-    } else {
-        ziti_log_lvl = level;
     }
 
     if (logger) {
         char msg[128];
-        int len = snprintf(msg, sizeof(msg), "set log level: ziti_log_lvl=%d &ziti_log_lvl = %p", ziti_log_lvl, &ziti_log_lvl);
+        int len = snprintf(msg, sizeof(msg), "set log level: %s=%d", marker ? marker : "root", level);
         logger(INFO, "ziti_log_set_level", msg, len);
     }
 }
 
-int ziti_log_level() {
+int ziti_log_level(const char *module, const char *file) {
+    int level;
+
+    file = basename(file);
```

Comment:
basename only returns NULL if there's an error (If `file` is passed in as NULL, "." is returned).  Is this what you're going for?

Reply:
> basename only returns NULL if there's an error (If `file` is passed in as NULL, "." is returned). Is this what you're going for?

we don't use standard basename() function